src/providers/dart_rename_provider.ts

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+import { window, workspace, RenameProvider, WorkspaceEdit, TextDocument, Position, CancellationToken, Uri, TextEdit, Range } from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+import * as as from "../analysis/analysis_server_types";
+
+export class DartRenameProvider implements RenameProvider {	
+	private analyzer: Analyzer;
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+	
+	provideRenameEdits(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {		
+		return workspace.saveAll(false).then(() => {
+			return this.doRename(document, position, newName, token);
+		});
+	}
+
+	private doRename(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {
+        return new Promise<WorkspaceEdit>((resolve, reject) => {
+			this.analyzer.editGetRefactoring({
+				kind: "RENAME",
+				file: document.fileName,
+				offset: document.offsetAt(position),
+				length: position.character,
+				validateOnly: false,
+				options: {
+					newName: newName
+				}			
+			}).then(resp => {				
+				let workspaceEdit = new WorkspaceEdit();
+
+				let hasError = this.handleProblem(
+					resp.initialProblems
+						.concat(resp.optionsProblems)
+						.concat(resp.finalProblems)
+				);
+
+				if (hasError) {
+					reject("Rename cancelled.");
```

Comment:
When there an error, messages such as below is shown.
![image](https://cloud.githubusercontent.com/assets/35133/17660700/62b13dec-630d-11e6-92c1-15fd914834ef.png)

If reject is called without parameter, the "Info" is still shown with message "No result". I rather have no "Info" message shown, just the "Error". I am not sure how to do this.